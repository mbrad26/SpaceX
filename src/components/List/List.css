.row {
  padding: 20px;
}

.card-title {
  font-size: 1.55rem;
}

.item-description {
  font-size: 0.9rem;
  overflow: hidden;
  display: -webkit-box;
  -webkit-line-clamp: 5;
  -webkit-box-orient: vertical;
}

.card-img-top {
  width: 100%;
  height: 35vh;
}
