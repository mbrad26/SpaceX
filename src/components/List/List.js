import React, { useContext } from 'react';
import CardGroup from "react-bootstrap/CardGroup";
import Card from "react-bootstrap/Card";
import Image from "react-bootstrap/Image";
import { Context } from '../App/App';
import './List.css';

const List = () => {
  const { data } = useContext(Context);

  return (
    <div className='row justify-content-around'>
      {data.map(
        item => <Item key={item.id} item={item} />
      )}
    </div>
  )
};

const Item = ({ item }) => {
  const { handleOpenModal } = useContext(Context);

  return (
    <>
      <CardGroup className="col-6 d-flex align-items-stretch container">
        <Card className="shadow">
          <Card.Img src={item.flickr_images[0]} alt='SpaceX' className="card-img-top" />
          <Card.Body>
            <Card.Title>
              {item.rocket_name ? item.rocket_name : item.name}
            </Card.Title>
            <Card.Text className="item-description">
              {item.description}
            </Card.Text>
            <a href="#" onClick={() => handleOpenModal(item)}>Details</a>
          </Card.Body>
        </Card>
      </CardGroup>
    </>
  )
};

export default List;
export { Item };
