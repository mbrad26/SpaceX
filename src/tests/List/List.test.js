import React from 'react';
import { render, screen } from '@testing-library/react';
import Context from '../../context/context';
import Provider from '../../context/provider';
import List from '../../components/List/List';
import { dataRockets } from '../fixtures';

describe('List', () => {
  it('renders snapshot', () => {
    const { container } = render(
      <Provider>
        <List />
      </Provider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders data', () => {
    const context = {
      data: dataRockets,
    }
    render(
      <Provider value={context}>
        <List />
      </Provider>
    );

    screen.debug();

    expect(screen.getByTestId('wrapper').children.length).toBe(2);
  });
});
