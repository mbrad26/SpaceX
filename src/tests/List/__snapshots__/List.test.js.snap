// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders snapshot 1`] = `undefined`;
